### PR TITLE
Add disablefirstlogin method

### PR DESCRIPTION
**Pull request by rahulsrivastava71**
Code is added to disable the first login page for VZ customer because currently the bootstrap admin password which shows on ui is not correct and this setting is created from rancher code itself , so to remove this better remove the first login page .

Provide a summary of the change and which issue (i.e. ticket) is fixed.
If there are any dependencies, for example on a PR in another repository, list those.

### CODE REVIEW

**Review by lake-of-dreams — CHANGES_REQUESTED** [review]
We only need to add logic for calling `disableFirstLogin` , pls remove all Keycloak auth related changes

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component.go [hunk 1]
```diff
@@ -257,6 +285,9 @@ func (r rancherComponent) PostInstall(ctx spi.ComponentContext) error {
 	if err := removeBootstrapSecretIfExists(log, c); err != nil {
 		return log.ErrorfThrottledNewErr("Error removing Rancher bootstrap secret: %s", err.Error())
 	}
+	if err := configureAuthProviders(ctx); err != nil {
```
Comment (bhatpmk) [inline platform-operator/controllers/verrazzano/component/rancher/rancher_component.go:288]: Is this change porting the Rancher-Keycloak integration to release-1.3 ?

Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/rancher/rancher_component.go:288]: yes

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component.go [hunk 2]
```diff
@@ -257,6 +256,9 @@ func (r rancherComponent) PostInstall(ctx spi.ComponentContext) error {
 	if err := removeBootstrapSecretIfExists(log, c); err != nil {
 		return log.ErrorfThrottledNewErr("Error removing Rancher bootstrap secret: %s", err.Error())
 	}
+	if err := disableFirstLogin(ctx); err != nil {
```
Comment (lake-of-dreams) [inline platform-operator/controllers/verrazzano/component/rancher/rancher_component.go:259]: This needs to be called from `PostUpgrade` as well..

Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/rancher/rancher_component.go:259]: where is this postupgrade method ? is it in upgrade_component.go

File: platform-operator/controllers/verrazzano/component/rancher/rancher.go [hunk 3]
```diff
@@ -50,6 +50,9 @@ const (
 	additionalTrustedCAsKey = "additionalTrustedCAs"
 	privateCAKey            = "privateCA"
 
+	KontainerDriverOKE = "oraclecontainerengine"
```
Comment (lake-of-dreams) [inline platform-operator/controllers/verrazzano/component/rancher/rancher.go:53]: why these are still here?

File: platform-operator/controllers/verrazzano/component/rancher/rancher.go [hunk 4]
```diff
@@ -67,8 +70,13 @@ const (
 	caCert                     = "ca.crt"
 	privateCAValue             = "true"
 	useBundledSystemChartValue = "true"
+	SettingFirstLogin          = "first-login"
 )
 
+var GVKSetting = common.GetRancherMgmtAPIGVKForKind("Setting")
+var GVKNodeDriver = common.GetRancherMgmtAPIGVKForKind("NodeDriver")
```
Comment (lake-of-dreams) [inline platform-operator/controllers/verrazzano/component/rancher/rancher.go:77]: you don't need GVKs for NodeDriver and KontainerDriver

File: platform-operator/controllers/verrazzano/component/rancher/rancher.go [hunk 5]
```diff
@@ -217,7 +225,7 @@ func deleteClusterRepos(log vzlog.VerrazzanoLogger) error {
 
 	config, err := ctrl.GetConfig()
 	if err != nil {
-		log.Debugf("Rancher IsReady: Failed getting config: %v", err)
+		log.Debugf("Rancher IsReaactivateOCIDriverdy: Failed getting config: %v", err)
```
Comment (lake-of-dreams) [inline platform-operator/controllers/verrazzano/component/rancher/rancher.go:228]: why is this changed?

File: tests/e2e/pkg/rancher.go [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+package pkg
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"strings"
+)
+
+// GvkToGvr converts a GroupVersionKind to corresponding GroupVersionResource
+func GvkToGvr(gvk schema.GroupVersionKind) schema.GroupVersionResource {
```
Comment (lake-of-dreams) [inline tests/e2e/pkg/rancher.go:12]: if only your test is using this function, just make it a part of that test file only, no need to separate out here if its nit being used elsewhere

File: tests/e2e/verify-infra/restapi/rancher_url_test.go [hunk 7]
```diff
@@ -150,10 +158,33 @@ var _ = t.Describe("rancher", Label("f:infra-lcm",
 
 					return httputil.ExtractFieldFromResponseBodyOrReturnError(string(body), "state", "unable to find state in Rancher clusters response")
 				}, waitTimeout, pollingInterval).Should(Equal("active"), "rancher local cluster not in active state")
+				verifySettingValue("first-login", "false", k8sClient)
```
Comment (lake-of-dreams) [inline tests/e2e/verify-infra/restapi/rancher_url_test.go:161]: call this after `metrics.emit`

File: tests/e2e/pkg/rancher.go (hunk 6 above)
Comment (lake-of-dreams) [reply tests/e2e/pkg/rancher.go:12]: or just leave it, will be easy to write any further tests if we had to

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component.go [hunk 8]
```diff
@@ -257,9 +256,27 @@ func (r rancherComponent) PostInstall(ctx spi.ComponentContext) error {
...
+		return log.ErrorfThrottledNewErr("failed disabling first login setting: %s", err.Error())
+	}
 	return r.HelmComponent.PostInstall(ctx)
 }
 
+// PostUpgrade configures the Rancher rest client and activates OCI and OKE drivers in Rancher
+func (r rancherComponent) PostUpgrade(ctx spi.ComponentContext) error {
```
Comment (lake-of-dreams) [inline platform-operator/controllers/verrazzano/component/rancher/rancher_component.go:266]: so `PostUpgrade` did not exist at all? I am just wondering if we should add it here at all if there was no `PostUpgrade` before?

File: platform-operator/controllers/verrazzano/component/rancher/rancher.go (hunk 3 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/rancher/rancher.go:53]: removed

File: platform-operator/controllers/verrazzano/component/rancher/rancher.go (hunk 4 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/rancher/rancher.go:77]: removed

File: platform-operator/controllers/verrazzano/component/rancher/rancher.go (hunk 5 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/rancher/rancher.go:228]: done

File: tests/e2e/pkg/rancher.go (hunk 6 above)
Comment (rahulsrivastava71) [reply tests/e2e/pkg/rancher.go:12]: sure

File: tests/e2e/verify-infra/restapi/rancher_url_test.go (hunk 7 above)
Comment (rahulsrivastava71) [reply tests/e2e/verify-infra/restapi/rancher_url_test.go:161]: done

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component.go (hunk 8 above)
Comment (rahulsrivastava71) [reply platform-operator/controllers/verrazzano/component/rancher/rancher_component.go:266]: no there was no post upgrade in 1.3 , but we can add as rancher setting is added in 1.3 so lets keep this here

File: tests/e2e/verify-infra/restapi/rancher_url_test.go [hunk 9]
```diff
@@ -37,7 +46,7 @@ var _ = t.Describe("rancher", Label("f:infra-lcm",
 				}
 
 				var rancherURL string
-
+				k8sClient, err := pkg.GetDynamicClientInCluster(kubeconfigPath)
```
Comment (lake-of-dreams) [inline tests/e2e/verify-infra/restapi/rancher_url_test.go:49]: it should fail if error in getting the k8sclient

Comment (rahulsrivastava71) [reply tests/e2e/verify-infra/restapi/rancher_url_test.go:49]: done

**Review by lake-of-dreams — DISMISSED** [review]

File: platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go [hunk 10]
```diff
@@ -284,13 +288,16 @@ func TestIsReady(t *testing.T) {
...
-//  THEN PostInstall should return nil
+//
+//	WHEN PostInstall is called
+//	THEN PostInstall should return nil
 func TestPostInstall(t *testing.T) {
 	// mock the k8s resources used in post install
 	caSecret := createCASecret()
 	rootCASecret := createRootCASecret()
 	adminSecret := createAdminSecret()
+	//authConfig := createKeycloakAuthConfig()
```
Comment (lake-of-dreams) [inline platform-operator/controllers/verrazzano/component/rancher/rancher_component_test.go:299]: remove this commented line

File: tests/e2e/pkg/rancher.go [hunk 11]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2021, 2022, Oracle and/or its affiliates.
```
Comment (lake-of-dreams) [inline tests/e2e/pkg/rancher.go:1]: change to `// Copyright (c) 2022, Oracle and/or its affiliates.`

Comment (rahulsrivastava71) [reply tests/e2e/pkg/rancher.go:1]: done

**Pull request merged**